docs/scarpet/api/Inventories.md

```diff
@@ -206,3 +206,138 @@ Returns size of the actual dropped items.
 inventory_drop(player(), 0, 1) => 1 // Q's one item on the ground
 inventory_drop(x,y,z, 0) => 64 // removed and spawned in the world a full stack of items
 </pre>
+
+## Screen handlers
+
+A screen handler is a value type used to open screens to a player and interact with them.
+For example, this includes the chest inventory gui, the crafting table gui and many more.
+
+### `create_screen(player, type, name, callback?)`
+
+Creates and opens a screen to a `player`.
+
+Available `type`s:
+
+* `anvil`
+* `beacon`
+* `brewing_stand`
+* `cartography_table`
+* `crafting`
+* `enchantment`
+* `furnace`
+* `generic_3x3`
+* `generic_9x1`
+* `generic_9x2`
+* `generic_9x3`
+* `generic_9x4`
+* `generic_9x5`
+* `generic_9x6`
+* `grindstone`
+* `hopper`
+* `lectern`
+* `loom`
+* `merchant`
+* `shulker_box`
+* `smithing`
+* `stonecutter`
+
+The `name` parameter can be a formatted text and will be displayed at the top of the screen.
+Some screens like the lectern or beacon screen don't show it.
+
+Optionally, a `callback` function can be passed as a fourth argument.
+This functions needs to have five parameters:
+`_(screen, player, action, index, button) -> ...`
```

Comment:
I smell, that technically screen and player are truly optional, because you have these references when creating the screen, right? but it actully might be useful when handling of the screen reference not to keep it globally and pass along.

Reply:
Yeah, you could save the screen globally, and since the player is passed to `create_screen` as well and doesn't change, it's available too. But it was much simpler do deal with, having them in the callback too. Especially with player scoped apps.